src/phoenix/server/api/mutations/playground_chat_mutations.py

```diff
@@ -0,0 +1,116 @@
+# mutations.py
+
+from collections.abc import Iterable, Iterator
+from typing import Any, Optional
+
+import strawberry
+from strawberry.types import Info
+from typing_extensions import TypeAlias, assert_never
+
+from phoenix.server.api.context import Context
+from phoenix.server.api.exceptions import BadRequest
+from phoenix.server.api.helpers.playground_clients import initialize_playground_clients
+from phoenix.server.api.helpers.playground_registry import PLAYGROUND_CLIENT_REGISTRY
+from phoenix.server.api.helpers.playground_spans import streaming_llm_span
+from phoenix.server.api.input_types.ChatCompletionInput import ChatCompletionInput
+from phoenix.server.api.input_types.TemplateOptions import TemplateOptions
+from phoenix.server.api.types.ChatCompletionMessageRole import ChatCompletionMessageRole
+from phoenix.server.api.types.ChatCompletionSubscriptionPayload import (
+    FinishedChatCompletion,
+)
+from phoenix.server.api.types.TemplateLanguage import TemplateLanguage
+from phoenix.server.dml_event import SpanInsertEvent
+from phoenix.utilities.template_formatters import (
+    FStringTemplateFormatter,
+    MustacheTemplateFormatter,
+    TemplateFormatter,
+)
+
+initialize_playground_clients()
+
+ChatCompletionMessage: TypeAlias = tuple[
+    ChatCompletionMessageRole, str, Optional[str], Optional[list[str]]
+]
+
+
+@strawberry.type
+class PlaygroundChatCompletionMutationMixin:
+    @strawberry.mutation
+    async def playground_chat_completion(
```

Comment:
there are a ton of playground-specific type definitions and I think our codebase has a lot of generic names that aren't actually reusable

Reply:
Yeah that makes a ton of sense was just curious but feel free to keep it!

Reply:
I think this should be `chat_completion` since it is part of the GraphQL API. A consumer of the GraphQL API could use this resolver to do a chat completion regardless of whether they are in playground.